go/vt/sqlparser/sql.y

```diff
@@ -3679,6 +3686,95 @@ over_opt:
     $$ = $1
   }
 
+frame_opt:
+  {
+    $$ = nil
+  }
+| ROWS frame_extent
+  {
+    $$ = &Frame{Unit: RowsUnit, Extent: $2}
+  }
+| RANGE frame_extent
+  {
+    $$ = &Frame{Unit: RangeUnit, Extent: $2}
+  }
+
+frame_extent:
+ BETWEEN frame_bound AND frame_bound
+    {
+       startBound := $2
+       endBound := $4
+       switch {
+       case startBound.Type == UnboundedFollowing:
+         yylex.Error("frame start cannot be UNBOUNDED FOLLOWING")
```

Comment:
There is a required directionality of FRAME: FOLLOWING < CURRENT ROW < PRECEEDING. Bounds that reverse those constraints are illegal.